portalnet/src/gossip.rs

```diff
@@ -243,84 +214,105 @@ pub async fn trace_propagate_gossip_cross_thread<TContentKey: OverlayContentKey>
     gossip_result
 }
 
-/// Filter all nodes from overlay routing table where XOR_distance(content_id, nodeId) < node radius
-fn calculate_interested_enrs<TContentKey: OverlayContentKey>(
-    content_key: &TContentKey,
-    all_nodes: &[&kbucket::Node<NodeId, Node>],
+/// Returns all ENRs that are connected and interested into provided content id.
+fn interested_enrs<TMetric: Metric>(
+    content_id: &[u8; 32],
+    kbuckets: Arc<RwLock<KBucketsTable<NodeId, Node>>>,
 ) -> Vec<Enr> {
-    let content_id = content_key.content_id();
-    // HashMap to temporarily store all interested ENRs and the content.
-    // Key is base64 string of node's ENR.
-
-    // Filter all nodes from overlay routing table where XOR_distance(content_id, nodeId) < node
-    // radius
-    let mut interested_enrs: Vec<Enr> = all_nodes
+    kbuckets
+        .write()
         .iter()
+        // Keep only connected nodes
+        .filter_map(|entry| {
+            if entry.status.is_connected() {
+                Some(entry.node)
+            } else {
+                None
+            }
+        })
+        // Keep only nodes that are interested in content
         .filter(|node| {
-            XorMetric::distance(&content_id, &node.key.preimage().raw()) < node.value.data_radius()
+            TMetric::distance(content_id, &node.key.preimage().raw()) <= node.value.data_radius
```

Comment:
I imagine we are inconsistent about `<=` and `=` for comparing to the radius. I just scanned the specs and don't see an official disambiguation. It's fine to change, because I think it's practically irrelevant. Do you think it's important? If so, maybe we should put it in the spec.

Reply:
In my mental model, radius was always inclusive. Reason being, if radius is MAX, we want all content, even if it is the exact opposite from our `node_id` (in practice, not likely to happen).

As I was typing this, I can see the argument for radius being non-inclusive. If radius is zero, we don't want any content, even if it is the same as our node_id. If I have to pick which of the two impossible scenarios make more sense, I would still go with the first one (you can interpret it as "I'm responsible for my node_id +/- radius", in which case, you are always "responsible" for at least your node_id because it's about you).

When deciding if we should store the content ([link](https://github.com/ethereum/trin/blob/b28beff3e83bc8bdcf7b9cf7bcb2f934bed652f0/trin-history/src/storage.rs#L32)) we use radius as inclusive, so at least now we will be somewhat consistent.

Either way, I think it's practically irrelevant, so I don't think it we need to update the spec. We can ask others, maybe somebody has strong opinion or good reason for one way or the other.